tutorials/scripting/gdextension/gdextension_cpp_example.rst

```diff
@@ -0,0 +1,643 @@
+.. _doc_gdextension_cpp_example:
+
+GDExtension C++ example
+====================
+
+Introduction
+------------
+
+The C++ bindings for GDExtension are built on top of the C GDExtension API
+and provide a nicer way to "extend" nodes in Godot using C++. This new system allows the 
+extension of Godot to nearly the same level as statically linked C++ modules.
+
+You can download the included example in the test folder of the godot-cpp repository `on
+GitHub <https://github.com/godotengine/godot-cpp>`__.
+
+Setting up the project
+----------------------
+
+There are a few prerequisites you'll need:
+
+- a Godot 4.x executable,
+- a C++ compiler,
+- SCons as a build tool,
+- a copy of the `godot-cpp
+  repository <https://github.com/godotengine/godot-cpp>`__.
+
+See also :ref:`Compiling <toc-devel-compiling>` as the build tools are identical
+to the ones you need to compile Godot from source.
+
+You can download these repositories from GitHub or let Git do the work for you.
+Note that these repositories have different branches for different versions
+of Godot. GDExtension modules written will not work in older versions of Godot (3.x) 
+and vice versa, so make sure you download the correct branch.
+
+.. note::
+
+    `GDExtension <https://godotengine.org/article/introducing-gd-extensions>`__
+    has been merged in the ``master`` branch of godot-cpp,
+    but it is only compatible with the upcoming Godot 4.0.
+    Therefore, you need to use the ``master`` branch of godot-cpp to use GDExtension
+    and follow this example.
+
+If you are versioning your project using Git, it is a good idea to add them as
+Git submodules:
+
+.. code-block:: none
+
+    mkdir gdextension_cpp_example
+    cd gdextension_cpp_example
+    git init
+    git submodule add -b master https://github.com/godotengine/godot-cpp
+    cd godot-cpp
+    git submodule update --init
+
+If you decide to just download the repositories or clone them into your project
+folder, make sure to keep the folder layout identical to the one described here,
+as much of the code we'll be showcasing here assumes the project follows this
+layout.
+
```

Comment:
What folder structure?
Maybe insert a graphic.

Reply:
Not sure what you mean. I have code blocks further below that showcase the file structure.